src/Ombi.Store/Repository/Requests/MovieRequestRepository.cs

```diff
@@ -46,12 +46,19 @@ public MovieRequests GetRequest(int theMovieDbId)
                 .FirstOrDefault();
         }
 
-        public IQueryable<MovieRequests> GetWithUser()
+        public IQueryable<MovieRequests> GetWithUser(bool anonimize)
         {
-            return Db.MovieRequests
+            var allRequests = Db.MovieRequests
                 .Include(x => x.RequestedUser)
                 .ThenInclude(x => x.NotificationUserIds)
-                .AsQueryable();
+                .ToList();
+
+            if (anonimize)
+            {
+                allRequests.ForEach(x => x.RequestedUser = null);
```

Comment:
We are iterating over the list of `allRequests` here twice, it would be better just to iterate over it once and set both properties to `null` in a single loop

Reply:
Isn't it so for LINQ that every chained method gets executed separately anyway, so executing the ForEach in the chain above would still result in double iteration? Not quite up to speed about the inner workings of LINQ though.

Reply:
Not sure I'm following, my suggestion would be,

```
foreach (var request in allRequests)
{
    request.RequestedUser = null;
    request.RequestedUserId = null;
}

```

That would be one iteration over the collection.


Looking a bit more in-depth, we could end up with a performance issue here. Originally the dataset would have been an `IQueryable`, and as part of EF Core that would mean that the query would not be evaluated on the database until some sort of result set is called e.g. `.ToList()`, before that happens that majority of methods calling this did further filtering on the MovieRequests, for example, paging or availability filtering, which reduced the results returned from the database for a quicker and less intensive query.

But now we are grabbing every single movie request from the database every time, then filtering an in-memory list, rather than building the query. If an Ombi setup has a few thousand movie requests (which lots do) then we are going to introduce a slowdown as we are asking the database for every request every time.

The solution to this would be to loop over the requests and null out the users after the query has been executed (so after a `.ToList()` for example). That would be a better solution here.

Reply:
You're totally right, i didn't factor in the fact i'm building a query, executing it and then converting it back to a query to be executed again. Will have a look at this in the coming days!